tests/test_backend_minio.py

```diff
@@ -0,0 +1,319 @@
+        fp.write(f"secret_key = {secret_key}\n")
+
+    backend = audbackend.backend.Minio(host, "repository")
+    assert backend.authentication == ("bad", "bad")
+    with pytest.raises(audbackend.BackendError):
+        backend.open()
+
+
+@pytest.mark.parametrize(
+    "interface",
+    [(audbackend.backend.Minio, audbackend.interface.Unversioned)],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "path, expected,",
+    [
+        ("/text.txt", "text/plain"),
+    ],
+)
+def test_content_type(tmpdir, interface, path, expected):
+    r"""Test setting of content type.
+
+    Args:
+        tmpdir: tmpdir fixture
+        interface: interface fixture
+        path: path of file on backend
+        expected: expected content type
+
+    """
+    tmp_path = audeer.touch(audeer.path(tmpdir, path[1:]))
+    interface.put_file(tmp_path, path)
+    stats = interface._backend._client.stat_object(interface.repository, path)
+    assert stats.content_type == expected
+
+
+@pytest.mark.parametrize(
+    "interface",
+    [(audbackend.backend.Minio, audbackend.interface.Unversioned)],
+    indirect=True,
+)
+@pytest.mark.parametrize(
+    "src_path, dst_path,",
+    [
+        (
+            "/big.1.txt",
+            "/big.2.txt",
+        ),
+    ],
+)
+def test_copy_large_file(tmpdir, interface, src_path, dst_path):
+    r"""Test copying of large files.
+
+    ``minio.Minio.copy_object()`` has a limit of 5 GB.
+    We mock the ``audbackend.backend.Minio._size()`` method
+    to return a value equivalent to 5 GB.
+    to trigger the fall back copy mechanism for large files,
+    without having to create a large file.
+
+    Args:
+        tmpdir: tmpdir fixture
+        interface: interface fixture
+        src_path: source path of file on backend
+        dst_path: destination of copy operation on backend
+
+    """
+    tmp_path = audeer.touch(audeer.path(tmpdir, "big.1.txt"))
+    interface.put_file(tmp_path, src_path)
+    interface._backend._size = lambda x: 5 * 1024 * 1024 * 1024
+    interface.copy_file(src_path, dst_path)
+    assert interface.exists(src_path)
+    assert interface.exists(dst_path)
+
+
+@pytest.mark.parametrize("host", [pytest.HOSTS["minio"]])
+@pytest.mark.parametrize("repository", [f"unittest-{pytest.UID}-{audeer.uid()[:8]}"])
+def test_create_delete_repositories(host, repository):
+    audbackend.backend.Minio.create(host, repository)
+    audbackend.backend.Minio.delete(host, repository)
+
+
+@pytest.mark.parametrize("host", [pytest.HOSTS["minio"]])
+@pytest.mark.parametrize("repository", [f"unittest-{pytest.UID}-{audeer.uid()[:8]}"])
+@pytest.mark.parametrize("authentication", [("bad-access", "bad-secret")])
+def test_errors(host, repository, authentication):
+    backend = audbackend.backend.Minio(host, repository, authentication=authentication)
+    with pytest.raises(audbackend.BackendError):
+        backend.open()
+
+
+def test_get_config(tmpdir, hosts, hide_credentials):
```

Comment:
This tests parsing of the configuration, but does not have a one-line docstring.

Reply:
I added a docstring.